ui/src/app/store/machine/utils/storage.ts

```diff
@@ -501,3 +501,39 @@ export const usesStorage = (fstype: Filesystem["fstype"] | null): boolean => {
   }
   return !["ramfs", "tmpfs"].includes(fstype);
 };
+
+/**
+ * Find the next available name for a storage device.
+ * @param disks - A machine's disks.
+ * @param prefix - A network interface type.
+ * @param nic - A network interface.
+ * @param vlan - A VLAN.
+ * @return An available name.
+ */
+export const getNextStorageName = (
+  disks: Disk[] | null,
+  prefix: "vg" | "md" | "bcache" | "datastore"
+): string | null => {
+  if (!disks) {
+    return null;
+  }
+  // Everything but datastores start their index at 0.
+  const startIndex = prefix === "datastore" ? 1 : 0;
+  // Get a list of the current names for the provided type.
+  const diskNames = disks.reduce<string[]>((names, disk) => {
+    if (
+      // Filter volume groups.
+      (prefix === "vg" && disk.type === DiskTypes.VOLUME_GROUP) ||
```

Comment:
Isn't there an `isVolumeGroup` util?

Reply:
Thanks, I looked for that and missed it somehow!